### DISCUSSION TITLE
Possible enhancement: graceful exit and reload

### DISCUSSION THREAD

**Discussion by mmendelson222** [WardBrian/mlb-led-scoreboard#13] [Ideas]
My first thought was, it would be nice to be able to exit the program without random pixels appearing on the screen.  If graceful exit of the rendering thread is possible, we could also load up a different configuration.  

My thought was to designate a file for this purpose, and have the rendering thread check the file in its loop (so it would exit at the end of a cycle).  The file would contain some sort of directive, like "exit", "load config-filename.json" or something similar.  For now those would be the two possibilities.

Any thoughts on this idea/approach, before I embark?

**Comment by WardBrian** [comment]
The exit should already be pretty graceful, the last thing the program always does is clear the screen. Has this not been your experience?

**Comment by mmendelson222** [comment]
I get pixels about 1/3 of the time when I run the program in the foreground and exit with ^C.  I'm looking at the code, but unsure where it's clearing the screen.

**Comment by WardBrian** [comment]
In main.py, the main loop is inside a global try/except/finally which calls matrix.clear before exiting when an exception occurs (Ctrl+C generates an exception in python so this should apply still)

**Comment by mmendelson222** [comment]
Ah, interesting.  Evidently ^C generates a KeyboardInterrupt, which is not an Exception.   I'll add that.

**Comment by smith288** [comment]
I just run a kill and look for the process in a bash script
